src/main/java/io/github/thebusybiscuit/slimefun4/core/guide/SlimefunGuideImplementation.java

```diff
@@ -70,7 +71,15 @@ default void unlockItem(Player p, SlimefunItem sfitem, Consumer<Player> callback
         if (p.getGameMode() == GameMode.CREATIVE && Slimefun.getRegistry().isFreeCreativeResearchingEnabled()) {
             research.unlock(p, true, callback);
         } else {
-            p.setLevel(p.getLevel() - research.getCost());
+            if (Slimefun.getCfg().getBoolean("researches.convert-level-cost-to-exp")) {
```

Comment:
We shouldn't read this from the config everytime a research is unlocked for performance reasons. Look into how it's handled with over config settings like the learning animation for example.

Reply:
fix